### PR TITLE
use `@branchHint` in `std.Random.float`

### PR DESCRIPTION
**Pull request by Rexicon226**
a quick TODO resolved with this shiny new builtin

### CODE REVIEW

**Comment by Rexicon226** [comment]
Ah yeah good point, feel free to cherry pick!

File: lib/std/Random.zig [hunk 1]
```diff
@@ -297,6 +296,7 @@ pub fn float(r: Random, comptime T: type) T {
                     const addl_rand_lz = @clz(r.int(u64));
                     rand_lz += addl_rand_lz;
                     if (addl_rand_lz != 64) {
+                        @brancHint(.likely);
```
Comment (SilasLock) [inline lib/std/Random.zig:299]: This looks like a minor typo; shouldn't it be `@branchHint`, not `@brancHint`?

Comment (Rexicon226) [reply lib/std/Random.zig:299]: Hah, thanks for spotting that. I’m blind.

Comment (SilasLock) [reply lib/std/Random.zig:299]: You're all good! Glad I could help! 😁

**Pull request merged**